mutable_batch/src/payload/partition.rs

```diff
@@ -762,20 +1077,34 @@ mod tests {
             assert_eq!(keys.len(), 1);
 
             // Reverse the encoding.
-            let reversed = build_column_values(&template, &keys[0]).map(|(k, v)| (k, v.to_string())).collect::<Vec<_>>();
+            let reversed: Vec<(&str, ColumnValue<'_>)> = build_column_values(&template, &keys[0]).collect();
 
             // Build the expected set of reversed tags by filtering out any
             // NULL tags (preserving empty string values).
-            let want_reversed = template.parts().filter_map(|v| match v {
+            let want_reversed: Vec<(&str, String)> = template.parts().filter_map(|v| match v {
                 TemplatePart::TagValue(col_name) if tag_values.contains_key(col_name) => {
                     // This tag had a (potentially empty) value wrote and should
                     // appear in the reversed output.
                     Some((col_name, tag_values.get(col_name).unwrap().to_string()))
                 }
                 _ => None,
-            }).collect::<Vec<_>>();
-
-            assert_eq!(reversed, want_reversed);
+            }).collect();
+
+            for (idx, want) in want_reversed.iter().enumerate() {
+                let got = &reversed[idx];
+
+                assert_eq!(got.0, want.0, "column names differ");
+
+                match got.1 {
+                    ColumnValue::Identity(_) => {
+                        // An identity is both equal to, and a prefix of, the
+                        // original value.
+                        assert_eq!(got.1, want.1, "identity values differ");
+                        assert!(got.1.is_prefix_match_of(&want.1), "prefix mismatch");
+                    },
+                    ColumnValue::Prefix(_) => assert!(got.1.is_prefix_match_of(&want.1), "prefix mismatch"),
```

Comment:
Thinking about debugging these tests if they fail, for the two `assert!`s here it'd be nice if the args to `is_prefix_match_of` were included in the failure message, something like:

```suggestion
                        assert!(got.1.is_prefix_match_of(&want.1), "prefix mismatch; {} is not a prefix of {}", got.1, want.1);
                    },
                    ColumnValue::Prefix(_) => assert!(got.1.is_prefix_match_of(&want.1), "prefix mismatch; {} is not a prefix of {}", got.1, want.1),
```